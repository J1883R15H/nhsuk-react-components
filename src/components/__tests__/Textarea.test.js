import React from 'react';
import { shallow } from 'enzyme';
import Textarea from '../form/components/textarea';
import Hint from '../hint';
import ErrorMessage from '../error-message';

describe('Textarea', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Textarea />);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
  it('handles onChange', () => {
    const valueCallback = jest.fn();
    const wrapper = shallow(<Textarea valueCallback={valueCallback} />);
    wrapper
      .find('textarea')
      .simulate('change', { target: { value: 'Change' } });
    expect(valueCallback).toHaveBeenCalled();
    expect(valueCallback.mock.calls[0][1]).toBe('Change');
    wrapper.unmount();
  });
  it('renders hint and error', () => {
    const wrapper = shallow(<Textarea error="Error" hint="Hint" />);
    expect(
      wrapper.containsMatchingElement(<ErrorMessage>Error</ErrorMessage>)
    ).toBeTruthy();
    expect(wrapper.containsMatchingElement(<Hint>Hint</Hint>)).toBeTruthy();
    wrapper.unmount();
  });
  it('has valid defaultProps', () => {
    expect(Textarea.defaultProps.valueCallback()).toBe(undefined);
    expect(Textarea.defaultProps.registerInitialValue()).toBe(undefined);
  });
});
