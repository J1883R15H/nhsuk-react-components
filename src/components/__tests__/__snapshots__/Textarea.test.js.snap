// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea matches snapshot 1`] = `
<Fragment>
  <textarea
    autoComplete=""
    className="nhsuk-textarea"
    onChange={[Function]}
    placeholder=""
    rows={5}
    style={Object {}}
  />
</Fragment>
`;
